.github/workflows/python-test.yml

```diff
@@ -87,35 +87,35 @@ jobs:
       - name: Show path of binaries
         shell: bash
         run: |
-          if [ "${{ matrix.version }}" = "15" -a "${{ matrix.os }}" = "windows-latest" ]; then
-            which clang-format
-            which clang-tidy
-          elif [ "${{ matrix.version }}" = "12.0.1"  -a "${{ matrix.os }}" = "ubuntu-latest" ]; then
-            which clang-format-12
-            which clang-tidy-12
-          else
-            which clang-format-${{ matrix.version }}
-            which clang-tidy-${{ matrix.version }}
+          if [ "${{ matrix.version }}" = "15" -o "${{ matrix.version }}" = "16" ] && [ "${{ matrix.os }}" = "windows-latest" ]; then
+              which clang-format
+              which clang-tidy
+          elif [ "${{ matrix.os }}" = "ubuntu-latest" ]; then
+              which "clang-format-${{ matrix.version }}"
+              which "clang-tidy-${{ matrix.version }}"
           fi
 
       - name: Check clang-tools on Windows
         if: matrix.os == 'windows-latest'
         shell: bash
         run: |
-          if [ "${{ matrix.version }}" = "15" ]; then
-            clang-format.exe --version
-            clang-tidy.exe --version
-          else
-            clang-format-${{ matrix.version }}.exe --version
-            clang-tidy-${{ matrix.version }}.exe --version
-          fi
+          case "${{ matrix.version }}" in
+            15|16)
+              clang-format.exe --version
+              clang-tidy.exe --version
+              ;;
+            *)
+              clang-format-${{ matrix.version }}.exe --version
+              clang-tidy-${{ matrix.version }}.exe --version
+              ;;
+          esac
 
       - name: Check clang-tools on Unix
         if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
         run: |
           if [ "${{ matrix.version }}" = "12.0.1" -a "${{ matrix.os }}" = "ubuntu-latest" ]; then
-            clang-format-12 --version
-            clang-tidy-12 --version
+            clang-format-12.0.1 --version
+            clang-tidy-12.0.1 --version
```

Comment:
I'm pretty sure if v12.0.1 is installed it can still be invoked with only the major version number. Meaning, I'm not sure this is needed. 

Also, clang v12.0.0 is no longer shipped with Github's Ubuntu runner, so there shouldn't be any concern for conflict there.

Reply:
This is concerning because cpp-linter only calls the binary with major version.

Reply:
Oh, I see from [this CI log](https://github.com/cpp-linter/clang-tools-pip/actions/runs/6859222837/job/18651217676#step:8:26) that the binary installed is named "clang-tidy-12.0.1`.

We should really get cpp-linter aligned to use binaries that this pkg installs/names. I suppose we could add other symlinks for clang-tidy-12 if the full version spec, 12.0.1, is specified. But that may cause regressive conflicts, like when 12.0.1 is specified and v12.0.1 is already installed.